### PR TITLE
Add codespell to pre-commit

### PR DESCRIPTION
**Pull request by ClausKlein**
- [x] depends on https://github.com/bemanproject/inplace_vector/pull/41

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -4,30 +4,37 @@ repos:
...
     # Markdown linting
     # Config file: .markdownlint.yaml
   - repo: https://github.com/igorshubovych/markdownlint-cli
     rev: v0.42.0
     hooks:
-    - id: markdownlint
+      - id: markdownlint
+
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.1.0
```
Comment (wusatosi) [inline .pre-commit-config.yaml:36]: Can we bump this to `v2.3.0`?

Comment (ClausKlein) [reply .pre-commit-config.yaml:36]: Copy and paste without checking the version 🥴

Comment (wusatosi) [reply .pre-commit-config.yaml:36]: That's what review is for :)

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -4,30 +4,37 @@ repos:
...
+    hooks:
+      - id: codespell
+        files: ^.*\.(cmake|cpp|hpp|txt|md|json|in|yaml|yml)$
+        args: ["--ignore-words", ".codespellignore" ]
```
Comment (wusatosi) [inline .pre-commit-config.yaml:40]: Just to confirm, this does not trigger auto correct right?

Comment (ClausKlein) [reply .pre-commit-config.yaml:40]: It shows only the issues and possible suggestions.

**Review by DeveloperPaul123 — APPROVED** [review]
LGTM, though I don't use `pre-commit`

**Comment by wusatosi** [comment]
> LGTM, though I don't use `pre-commit`

I do recommend you set it up if you plan to contribute to this project.
Otherwise CI will be angry.

**Comment by DeveloperPaul123** [comment]
> > LGTM, though I don't use `pre-commit`
> 
> I do recommend you set it up if you plan to contribute to this project. Otherwise CI will be angry.

I might, we'll see. I generally don't maintain a global python install as I use `uv` so I'm not the biggest fan of tools that must be installed via `pip`. I'll see if I can give it a try though.

**Pull request merged**